ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/IntegratedWindowsAuthentication/IntegratedWindowsAuthentication.cpp

```diff
@@ -0,0 +1,209 @@
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<IntegratedWindowsAuthentication>("Esri.Samples", 1, 0, "IntegratedWindowsAuthenticationSample");
+}
+
+MapQuickView* IntegratedWindowsAuthentication::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void IntegratedWindowsAuthentication::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void IntegratedWindowsAuthentication::searchPublicPortal()
+{
+  if (!m_publicPortal)
+  {
+    m_publicPortal = new Portal(arcgis_url, this);
+  }
+  else
+  {
+    m_loadingIndicator = true;
+    emit isLoadingChanged();
+    m_publicPortal->findItems(*query);
+    return;
+  }
+
+  connect(m_publicPortal, &Portal::doneLoading, this, [this](Error loadError)
+  {
+    if (!loadError.isEmpty())
+    {
+      m_mapLoadError = loadError.message();
+      m_loadingIndicator = false;
+      emit isLoadingChanged();
+      emit mapLoadErrorChanged();
+      return;
+    }
+
+    m_publicPortal->findItems(*query);
+  });
+
+  connect(m_publicPortal, &Portal::findItemsCompleted, this, [this](PortalQueryResultSetForItems* result)
+  {
+    if(!result)
+      return;
+
+    if(m_webmaps)
+    {
+      m_webmaps = nullptr;
+    }
+
+    m_webmaps = result->itemResults();
+
+     emit webmapListModelChanged();
+     m_loadingIndicator = false;
+     emit isLoadingChanged();
+  });
+
+  m_loadingIndicator = true;
+  emit isLoadingChanged();
+
+  m_publicPortal->load();
+}
+
+void IntegratedWindowsAuthentication::searchIwaSecurePortal(const QString& url)
+{
+  if (!m_iwaSecurePortal)
+  {
+    m_iwaSecurePortal = new Portal(url, this);
+  }
+  else if (m_iwaSecurePortal && (m_iwaSecurePortal->loadStatus() == LoadStatus::FailedToLoad))
+  {
+    m_iwaSecurePortal = nullptr;
+    m_iwaSecurePortal = new Portal(url, this);
+  }
+  else if (m_iwaSecurePortal && (m_iwaSecurePortal->loadStatus() == LoadStatus::Loaded))
+  {
+    if (m_iwaSecurePortal->url() != url)
+    {
+      m_iwaSecurePortal = nullptr;
+      m_iwaSecurePortal = new Portal(url, this);
+    }
+    else
+    {
+      m_loadingIndicator = true;
+      emit isLoadingChanged();
+      m_iwaSecurePortal->findItems(*query);
+      return;
+    }
+  }
+
+  connect(m_iwaSecurePortal, &Portal::doneLoading, this, [this](Error loadError)
+  {
+    if (!loadError.isEmpty())
+    {
+      m_mapLoadError = loadError.message();
+      m_loadingIndicator = false;
+
+      if(m_webmaps)
+        m_webmaps = nullptr;
+
+      emit webmapListModelChanged();
+      emit isLoadingChanged();
+      emit mapLoadErrorChanged();
+      return;
+    }
+
+    m_iwaSecurePortal->findItems(*query);
+  });
+
+  connect(m_iwaSecurePortal, &Portal::findItemsCompleted, this, [this](PortalQueryResultSetForItems* result)
```

Comment:
This is pretty much the same code as the public portal's signal handler. Since you need to connect several objects to the same handler, a better option would probably be to either create a SLOT that is called by each, or to take out the logic from each and put it in a separate member function, and call the method from the lambda. 

- Here is how you could create a SLOT and hook it up - https://github.com/Esri/arcgis-runtime-samples-qt/blob/7e53f3d4f69a063cfa08ba092d0732d9f72b0c12/ArcGISRuntimeSDKQt_CppSamples/Scenes/OpenMobileScenePackage/OpenMobileScenePackage.cpp#L164

- alternatively, you could create a member function called `processResults` and call that from here, and stick all this logic in there. That should eliminate the duplication.

Same concept probably applies to the done loading signal handler